src/commit.c

```diff
@@ -428,11 +441,16 @@ int git_commit__parse(void *_commit, git_odb_object *odb_obj)
 	}
 
 	/* Always parse the committer; we need the commit time */
-	commit->committer = git__malloc(sizeof(git_signature));
-	GITERR_CHECK_ALLOC(commit->committer);
+	if (flags & (GIT_COMMIT_PARSE_ALL|GIT_COMMIT_PARSE_QUICK)) {
```

Comment:
Same here -- no need for the conditional if you do so for all available flags anyway.